src/main/java/Circle.java

```diff
@@ -0,0 +1,20 @@
+public record Circle(Point point, double radius) implements Shape {
+    public Circle {
+        if (radius < 0)
+            throw new IllegalArgumentException("Radius shouldn't be negative.");
+    }
+
+    @Override
+    public String getArea() throws IllegalStateException {
+        String resultMsg = "";
+        if (this.point == null)
```

Comment:
to powinno być sprawdzane już na etapie konstruktora - bo w tym momencie to już musztarda po obiedzie :)

Reply:
słuszna uwaga, dzięki - przeniosłam do konstruktora

Reply:
słuszna uwaga, dzięki - przeniosłam do konstruktora